lib/osf-components/addon/components/file-actions-menu/submit-to-boa-modal/component.ts

```diff
@@ -0,0 +1,112 @@
+import { inject as service } from '@ember/service';
+import { waitFor } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+import { alias } from '@ember/object/computed';
+import { task } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import IntlService from 'ember-intl/services/intl';
+import File from 'ember-osf-web/packages/files/file';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+import config from 'ember-osf-web/config/environment';
+import StorageManager from 'osf-components/components/storage-provider-manager/storage-manager/component';
+
+interface Args {
+    file: File;
+    isOpen: boolean;
+    closeModal: () => {};
+    manager: StorageManager;
+    onSubmitToBoa: () => {};
+}
+
+export default class SubmitToBoaModal extends Component<Args> {
+    @service toast!: Toastr;
+    @service intl!: IntlService;
+    datasets?: string[];
+    @alias('loadDatasetsTask.isRunning') loadDatasetsRunning!: boolean;
+
+    @task
+    @waitFor
+    async loadDatasetsTask() {
+        const datasets = [
+            '2022 Jan/Java',
+            '2022 Feb/Python',
+            '2021 Method Chains',
+            '2021 Aug/Python',
+            '2021 Aug/Kotlin (small)',
+            '2021 Aug/Kotlin',
+            '2021 Jan/ML-Verse',
+            '2020 August/Python-DS',
+            '2019 October/GitHub (small)',
+            '2019 October/GitHub (medium)',
+            '2019 October/GitHub',
+            '2015 September/GitHub',
+            '2013 September/SF (small)',
+            '2013 September/SF (medium)',
+            '2013 September/SF',
+            '2013 May/SF',
+            '2013 February/SF',
+            '2012 July/SF',
+        ];
+
+        this.set('datasets', datasets.toArray());
+    }
+
+    @task
+    @waitFor
+    async confirmSubmitToBoa() {
+        try {
+            const dataset = $('#boa_datasets')[0].value;
+            const manager = this.args.manager;
+            const file = this.args.file;
+            const fileModel = file.fileModel;
+            const endpoint = config.OSF.url + 'api/v1/project/' + fileModel.target.get('id') + '/boa/submit-job/';
+            const payload = {
+                data: {
+                    nodeId: fileModel.target.get('id'),
+                    name: file.name,
+                    materialized: fileModel.materializedPath,
+                    links: {
+                        download: file.links.download,
+                        upload: file.links.upload,
+                    },
+                },
+                parent: {
+                    links: {
+                        upload: manager.currentFolder.links.upload,
```

Comment:
If we are trying to do a method that doesn't rely on a `manager` being passed in, I _think_ this can just be `upload: fileModel.parentFolder.get('links').upload`